lib/ets/bag.ex

```diff
@@ -353,14 +353,16 @@ defmodule ETS.Bag do
   """
   @spec match(Bag.t(), ETS.match_pattern(), non_neg_integer()) ::
           {:ok, {[tuple()], any() | :end_of_table}} | {:error, any()}
-  def match(%Bag{table: table}, pattern, limit), do: Base.match(table, pattern, limit)
+  def match(%Bag{table: table}, pattern, limit) when is_atom(pattern) or is_tuple(pattern),
+    do: Base.match(table, pattern, limit)
```

Comment:
match/2 includes these guards so I added them to match/3 also for consistency.  Do we still want these guards?  If not they could be removed across the board.